lib/broadway_cloud_pub_sub/google_api_client.ex

```diff
@@ -136,20 +136,38 @@ defmodule BroadwayCloudPubSub.GoogleApiClient do
   defp handle_response({:ok, _}, _), do: :ok
 
   defp handle_response({:error, reason}, :receive_messages) do
-    Logger.error("Unable to fetch events from Cloud Pub/Sub. Reason: #{inspect(reason)}")
+    Logger.error("Unable to fetch events from Cloud Pub/Sub. Reason: #{inspect_error(reason)}")
     []
   end
 
   defp handle_response({:error, reason}, :acknowledge) do
-    Logger.error("Unable to acknowledge messages with Cloud Pub/Sub, reason: #{inspect(reason)}")
+    Logger.error(
+      "Unable to acknowledge messages with Cloud Pub/Sub, reason: #{inspect_error(reason)}"
+    )
+
     :ok
   end
 
   defp handle_response({:error, reason}, :put_deadline) do
-    Logger.error("Unable to put new ack deadline with Cloud Pub/Sub, reason: #{inspect(reason)}")
+    Logger.error(
+      "Unable to put new ack deadline with Cloud Pub/Sub, reason: #{inspect_error(reason)}"
+    )
+
     :ok
   end
 
+  defp inspect_error(%Tesla.Env{} = reason) do
+    # (__client__ includes the access token)
+    reason
+    |> Map.from_struct()
+    |> Map.delete(:__client__)
+    |> inspect()
+  end
```

Comment:
Would we be better off to provide a more readable (albeit less verbose) error message?

```suggestion
  defp inspect_error(%Tesla.Env{} = env) do
    "Request to #{env.url} failed with status #{env.status}"
  end
```

Reply:
I think there'd be value in including the response body as well.

I first saw this with an error unrelated to auth, and having the response body was helpful in debugging my problem.

Thoughts?

Reply:
Agreed!  Maybe something like this?

```ex
"""
Request to #{env.url} failed with status #{env.status}, got:

#{env.body}

"""

Reply:
@mcrumm Updated!  I added some `inspect`'s just to be safe

<img width="920" alt="nvim" src="https://user-images.githubusercontent.com/446024/110805536-2afa3500-824f-11eb-8875-6ad2a1449594.png">